source/Octostache.Tests/FiltersFixture.cs

```diff
@@ -90,6 +92,109 @@ public void JsonIsEscaped(string input, string expectedResult, string testName)
+        [InlineData("single'quote", "single'quote")]
+        [InlineData("我叫章鱼", "\\u6211\\u53eb\\u7ae0\\u9c7c")]
+        [InlineData(null, "")]
+        public void YamlDoubleQuoteIsEscaped(string input, string expectedResult)
+        {
+            var result = Evaluate("#{Foo | YamlDoubleQuoteEscape}", new Dictionary<string, string> { { "Foo", input } });
+            result.Should().Be(expectedResult);
+        }
+
+        private class TestDocument
+        {
+            public string Key { get; set; }
+        }
+        
+        [Theory]
+        [InlineData("")]
+        [InlineData("a")]
+        [InlineData("\"")]
+        [InlineData("'")]
+        [InlineData("\t")]
+        [InlineData("\n")]
+        [InlineData("\r\n")]
+        [InlineData("我叫章鱼")]
+        [InlineData("This\nis a more\r\n \"complicated\"\texample \\❤")]
+        public void YamlDoubleQuotedStringsCanRoundTrip(string input)
+        {
+            var yaml = Evaluate("Key: \"#{Input | YamlDoubleQuoteEscape}\"", new Dictionary<string, string> { { "Input", input } });
+
+            var doc = new DeserializerBuilder()
+                .Build()
+                .Deserialize<TestDocument>(yaml);
+
+            doc.Key.Should().Be(input);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("a")]
+        [InlineData("\"")]
+        [InlineData("'")]
+        [InlineData("\t")]
+        [InlineData("\n")]
+        [InlineData("我叫章鱼")]
+        [InlineData("this contains six spaces\nand one line break")]
+        public void YamlSingleQuotedStringsCanRoundTrip(string input)
+        {
+            var yaml = Evaluate("Key: '#{Input | YamlSingleQuoteEscape}'", new Dictionary<string, string> { { "Input", input } });
+
+            var doc = new DeserializerBuilder()
+                .Build()
+                .Deserialize<TestDocument>(yaml);
+
+            doc.Key.Should().Be(input);
+        }
+
+        [Theory]
+        [InlineData("\r\n", "\r\n")]
+        [InlineData("\r\n\r\n", "\r\n\r\n")]
+        [InlineData("a\nb", "a\nb")]
+        [InlineData("a \nb", "a\nb")]
+        [InlineData("a\n\nb", "a\n\nb")]
+        [InlineData("a\r\nb", "a\r\nb")]
+        [InlineData("a\r\n\nb", "a\r\n\nb")]
+        [InlineData("a \n\nb", "a\n\nb")]
+        [InlineData("a\n\n\n\n\r\nb", "a\n\n\n\n\r\nb")]
+        [InlineData("this contains six spaces\nand one line break", "this contains six spaces\nand one line break")]
+        public void YamlSingleQuotedStringsCanRoundTripWithSideEffects(string input, string expected)
+        {
+            var yaml = Evaluate("Key: '#{Input | YamlSingleQuoteEscape}'", new Dictionary<string, string> { { "Input", input } });
+
+            var doc = new DeserializerBuilder()
+                .Build()
+                .Deserialize<TestDocument>(yaml);
+
+            // Yamldotnet normalises \r\n in single quoted scalars to \n.
+            var normalisedExpected = expected.Replace("\r\n", "\n");
```

Comment:
This is the best approach I could find. I've left the `expected` strings as they are (because they represent what we expect), and then performed a replace on them down here because if I'm interpreting it right, the normalisation of the newlines is a decision made by Yamldotnet, and not a strict requirement of the yaml spec.